src/store/uiState.js

```diff
@@ -0,0 +1,74 @@
+const localStorageKey = "uiState";
+const version = 1;
+let initialize = false;
+
+const getDefaultState = () => {
+	return {
+		pagination: {},
+		filter: {},
+		version,
+	};
+};
+
+export const actions = {};
+
+export const getters = {
+	getStateByKey: (state) => ({ key, identifier }) => {
+		if (initialize) {
+			if (key) {
+				if (identifier) {
+					return state[key][identifier];
+				} else {
+					return state[key];
+				}
+			} else {
+				throw new SyntaxError("Key is missing!");
+			}
+		} else {
+			throw new SyntaxError("uiState not initialize");
+		}
```

Comment:
You could escape the condition hell here, by using early error handling.
```suggestion
		if (!initialize) {
			throw new SyntaxError("uiState not initialize");
		}
		if (!key) {
			throw new SyntaxError("Key is missing!");
		}
		return identifier
			? state[key][identifier]
			: state[key]
```